codefori/vscode-ibmi
# Add support for OpenJDK 11 to run the Debug Service

### Changes
It's possible to install OpenJDK 11 using yum, which is an alternative to using IBM's JDK.
This PR brings the required changes to support OpenJDK 11 to run the Debug Service:
- Java installations are now connection features
- When looking for a Java version, if the IBM version is not found, the lookup falls back to the openJDK version if available
- The certificate generation process backups and changes the shell scripts provided with the Debug Service to make them compatible with OpenJDK 11

### How to test this PR
On a system with Debug Service 2 and no IBM JDK 11 installed, before testing the PR:
1. Stop the Debug Service
2. Install OpenJDK 11 with yum: `yum install openjdk-11-ea.ppc64`

Then:
1. Connect to the system (Code for i should detect available Java installations)
2. Go to the IBM i Debugger view
3. Re-generate the certificate
4. Start the debug service

### Checklist
* [x] have created one or more test cases (for java locations)
